odoo/addons/base/tests/test_test_suite.py

```diff
@@ -10,3 +20,385 @@ class TestTestSuite(TestCase):
+            def handle(logger, record): # disable error logging
+                return
+            fake_result = OdooTestResult()
+            with patch('logging.Logger.makeRecord', makeRecord), patch('logging.Logger.handle', handle):
+                super()._feedErrorsToResult(fake_result, errors)
+            self.check_log_records(log_records)
+        except Exception as e:
+            _logger.exception(e)  # we don't expect _feedErrorsToResult to raise any exception, this will make it more robust to future change and eventual mistakes
+
+    def check_log_records(self, log_records):
+        for log in log_records:
+            self.assert_log_equal(log, 'logger', _logger)
+            self.assert_log_equal(log, 'name', 'odoo.addons.base.tests.test_test_suite')
+            self.assert_log_equal(log, 'fn', __file__)
+            self.assert_log_equal(log, 'func', self._testMethodName)
+
+        self.catched_logs += log_records
+        if self.expected_logs is not None:
+            for log in log_records:
+                level, msg = self.expected_logs.pop(0)
+                self.assert_log_equal(log, 'level', level)
+                self.assert_log_equal(log, 'msg', msg, self.clean_message)
+
+    def assert_log_equal(self, log, key, expected, cleanup=(lambda x: x)):
+        value = cleanup(log[key])
+        if value != expected:
+            log_str = "" if key == 'msg' else f"\n{log['msg']}"
+            self.log_error(f"Key `{key}` => `{value}` is not equal to `{expected}`.{log_str}")
+
+    def log_error(self, message):
+        # we would just log, but using the test_result will help keeping the tests counters corrects.
+        self.test_result.addError(self, (AssertionError, AssertionError(message), None))
+
+    def clean_message(self, message):
+        root_path = __file__.replace('/odoo/odoo/addons/base/tests/test_test_suite.py', '')
+        message = re.sub(r'line \d+', 'line $line', message)
+        message = re.sub(r'py:\d+', 'py:$line', message)
+        message = message.replace(f'{root_path}', '/root_path')
+        return message
+
+
+class TestRunnerLogging(TestRunnerLoggingCommon):
+
+    def test_raise(self):
+        raise Exception('This is an error')
+
+    def test_raise_subtest(self):
+        """
+        with subtest, we expect to have multiple errors, one per subtest
+        """
+        def make_message(message):
+            return (
+f'''ERROR: Subtest TestRunnerLogging.test_raise_subtest (<subtest>)
+Traceback (most recent call last):
+  File "/root_path/odoo/odoo/addons/base/tests/test_test_suite.py", line $line, in test_raise_subtest
+    raise Exception('{message}')
+Exception: {message}
+''')
+        self.expected_logs = [
+            (20, '=' * 70),
+            (40, make_message('This is an error')),
+        ]
+        with self.subTest():
+            raise Exception('This is an error')
+
+        self.assertFalse(self.expected_logs, "Error should have been logged immediatly")
```

Comment:
This is explained in the TestRunnerLoggingCommon doc

    """
    Metatesting: check that on error, the runner logged in the right file.
    This is mainly to avoid having errors in test/common.py or test/runner.py`
    This kind of metatesting can be tricky, in this case the logs are made outside of the test method,
    after the teardown actually.
    """